### PR TITLE
feat: add eslint & fix cli-ux deps

### PR DESCRIPTION
**Pull request by mo4islona**

### CODE REVIEW

File: cli/src/commands/squid/create.ts [hunk 1]
```diff
@@ -1,47 +1,47 @@
 import { Command, Flags } from '@oclif/core';
-import { create } from '../../rest-client/routes/create';
+import { create } from '../../rest-client';
 
 export default class Create extends Command {
-    static description = 'Create a squid';
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:5]: Create a Squid

File: cli/src/commands/squid/create.ts [hunk 2]
```diff
@@ -1,47 +1,47 @@
...
-    static args = [
-        {
-            name: 'name',
-            description: 'squid name',
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:9]: Squid name

File: cli/src/commands/squid/create.ts [hunk 3]
```diff
@@ -1,47 +1,47 @@
...
-            required: true,
-        },
-    ];
+  static description = 'Create a squid';
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:5]: Create a Squid

File: cli/src/commands/squid/create.ts [hunk 4]
```diff
@@ -1,47 +1,47 @@
...
+  static args = [
+    {
+      name: 'name',
+      description: 'squid name',
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:9]: Squid name

File: cli/src/commands/squid/create.ts [hunk 5]
```diff
@@ -1,47 +1,47 @@
...
 
-    static flags = {
-        description: Flags.string({
-            char: 'd',
-            description: 'description',
-            required: false,
-        }),
-        logo: Flags.string({
-            char: 'l',
-            description: 'logo url',
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:22]: logo URL

File: cli/src/commands/squid/create.ts [hunk 6]
```diff
@@ -1,47 +1,47 @@
...
-            required: false,
-        }),
-        website: Flags.string({
-            char: 'w',
-            description: 'website url',
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:27]: website URL

File: cli/src/commands/squid/create.ts [hunk 7]
```diff
@@ -1,47 +1,47 @@
...
-    };
+  static flags = {
+    description: Flags.string({
+      char: 'd',
+      description: 'description',
+      required: false,
+    }),
+    logo: Flags.string({
+      char: 'l',
+      description: 'logo url',
```
Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:22]: logo URL

Comment (DLangellotti) [inline cli/src/commands/squid/create.ts:22]: logo URL

File: cli/src/commands/squid/kill.ts [hunk 8]
```diff
@@ -1,33 +1,33 @@
...
-    destroyApp,
-    destroyDeployment,
+  destroyApp,
+  destroyDeployment,
 } from '../../rest-client/routes/destroy';
-import { parseNameAndVersion } from '../../utils'
+import { parseNameAndVersion } from '../../utils';
 
 export default class Kill extends Command {
-    static description = 'Kill a squid or a squid version';
```
Comment (DLangellotti) [inline cli/src/commands/squid/kill.ts:9]: Kill a Squid or a version of a Squid

File: cli/src/commands/squid/kill.ts [hunk 9]
```diff
@@ -1,33 +1,33 @@
...
-    static args = [
-        {
-            name: 'nameAndVersion',
-            description: '<name> or <name@version>',
-            required: true,
-        },
-    ];
+  static description = 'Kill a squid or a squid version';
```
Comment (DLangellotti) [inline cli/src/commands/squid/kill.ts:9]: Kill a Squid or a version of a Squid

File: cli/src/commands/squid/ls.ts [hunk 10]
```diff
@@ -1,57 +1,60 @@
...
 import { versionList } from '../../rest-client/routes/versions';
 
 export default class Ls extends Command {
-    static description = 'Squid or versions list';
+  static description = 'Squid or versions list';
 
-    static flags = {
-        name: Flags.string({
-            char: 'n',
-            description: 'squid name',
```
Comment (DLangellotti) [inline cli/src/commands/squid/ls.ts:12]: Squid name

File: cli/src/commands/squid/ls.ts [hunk 11]
```diff
@@ -1,57 +1,60 @@
...
-            char: 't',
-            description: 'truncate data in columns: false by default',
-            required: false,
-            default: false,
-        }),
-    };
+  static flags = {
+    name: Flags.string({
+      char: 'n',
+      description: 'squid name',
```
Comment (DLangellotti) [inline cli/src/commands/squid/ls.ts:11]: Squid name

File: cli/src/commands/squid/release.ts [hunk 12]
```diff
@@ -2,79 +2,79 @@ import { Command, Flags } from '@oclif/core';
...
 const options: Partial<SimpleGitOptions> = {
-    baseDir: process.cwd(),
-    binary: 'git',
+  baseDir: process.cwd(),
+  binary: 'git',
 };
 const git: SimpleGit = simpleGit(options);
 
 export default class Release extends Command {
-    static description = 'Create a new Squid version';
```
Comment (DLangellotti) [inline cli/src/commands/squid/release.ts:17]: Create a new version of a Squid

File: cli/src/commands/squid/release.ts [hunk 13]
```diff
@@ -2,79 +2,79 @@ import { Command, Flags } from '@oclif/core';
...
-    static args = [
-        {
-            name: 'nameAndVersion',
-            description: 'name@version',
-            required: true,
-        },
-    ];
+  static description = 'Create a new Squid version';
```
Comment (DLangellotti) [inline cli/src/commands/squid/release.ts:17]: Create a new version of a Squid

File: cli/src/commands/squid/release.ts [hunk 14]
```diff
@@ -2,79 +2,79 @@ import { Command, Flags } from '@oclif/core';
...
+      name: 'nameAndVersion',
+      description: 'name@version',
+      required: true,
+    },
+  ];
 
-    static flags = {
-        source: Flags.string({
-            char: 's',
-            description: 'A fully qualified git url, e.g. https://github.com/squidlover/my-squid.git#v5',
```
Comment (DLangellotti) [inline cli/src/commands/squid/release.ts:29]: a fully qualified git URL

File: cli/src/commands/squid/release.ts [hunk 15]
```diff
@@ -2,79 +2,79 @@ import { Command, Flags } from '@oclif/core';
...
-            squidName,
-            versionName,
-            deployUrl as string,
-            description
-        );
-        this.log(
-            '◷ You may now detach from the build process by pressing Ctrl + C. The Squid deployment will continue uninterrupted.'
-        );
-        this.log(
-            '◷ The new squid will be available as soon as the deployment is complete.'
```
Comment (DLangellotti) [inline cli/src/commands/squid/release.ts:70]: The new Squid...

File: cli/src/rest-client/request.ts [hunk 16]
```diff
@@ -1,28 +1,65 @@
...
+    for (const error of responseBody.errors) {
+      for (const constraint of Object.values(error.constraints)) {
+        validationErrorString += `${constraint}\n`;
+      }
     }
+    throw new Error(validationErrorString);
+  } else if (response.status === 200) {
+    return { status: response.status, body: responseBody };
+  } else {
+    throw new Error(`Squid server error. Please come back later. If the error persists please open an issue at https://github.com/subsquid/squid and report to t.me/HydraDevs`);
```
Comment (DLangellotti) [inline cli/src/rest-client/request.ts:63]: If the error persists, open

File: cli/src/rest-client/routes/me.ts [hunk 17]
```diff
@@ -1,15 +1,9 @@
...
-        },
-    });
-    const responseBody = await response.json();
-    if (response.status === 200) {
-        return `Successfully logged as ${responseBody.username}`;
-    }
+export async function me(): Promise<string | undefined> {
+  const { status, body } = await request(`${baseUrl}/client/me`);
+  if (status === 200) {
+    return `Successfully logged as ${body.username}`;
```
Comment (DLangellotti) [inline cli/src/rest-client/routes/me.ts:7]: Successfully logged in as

File: cli/src/utils.ts [hunk 18]
```diff
@@ -1,176 +1,175 @@
...
-                            )
-                        );
-                    }
-                    break;
-                case DeployPipelineStatusEnum.IMAGE_BUILDING:
-                    cli.action.start('◷ Building your squid');
-                    if (pipeline.isErrorOccurred) {
-                        command.error(
-                            buildPipelineErrorMessage(
-                                `❌ An error occurred during building process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:51]: ...during the building process

File: cli/src/utils.ts [hunk 19]
```diff
@@ -1,176 +1,175 @@
...
-                            )
-                        );
-                    }
-                    break;
-                case DeployPipelineStatusEnum.IMAGE_PUSHING:
-                    cli.action.start('◷ Publishing your squid');
-                    if (pipeline.isErrorOccurred) {
-                        command.error(
-                            buildPipelineErrorMessage(
-                                `❌ An error occurred during pushing process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:62]: An error occurred during the pushing process

File: cli/src/utils.ts [hunk 20]
```diff
@@ -1,176 +1,175 @@
...
-                            )
-                        );
-                    }
-                    break;
-                case DeployPipelineStatusEnum.DEPLOYING:
-                    cli.action.start('◷ Almost ready');
-                    if (pipeline.isErrorOccurred) {
-                        command.error(
-                            buildPipelineErrorMessage(
-                                `❌ An error occurred during deploying process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:73]: An error occurred during the deployment process

File: cli/src/utils.ts [hunk 21]
```diff
@@ -1,176 +1,175 @@
...
-                                pipeline.comment
-                            )
-                        );
-                    }
-                    break;
-                case DeployPipelineStatusEnum.OK:
-                    command.log(
-                        `◷ Your squid almost ready and will be accessible on ${deploymentUrl}`
```
Comment (DLangellotti) [inline cli/src/utils.ts:81]: Your Squid is almost ready and will be accessible at

File: cli/src/utils.ts [hunk 22]
```diff
@@ -1,176 +1,175 @@
...
+  while (inProgress) {
+    const pipeline = await getDeployPipeline(squidName, versionName);
+    if (pipeline) {
+      if (pipeline.status !== lastStatus) {
+        lastStatus = pipeline.status;
+        CliUx.ux.action.stop('✔️');
+      }
+      switch (pipeline?.status) {
+      case DeployPipelineStatusEnum.CREATED:
+        CliUx.ux.action.start('◷ Preparing your squid');
```
Comment (DLangellotti) [inline cli/src/utils.ts:35]: Preparing your Squid

File: cli/src/utils.ts [hunk 23]
```diff
@@ -1,176 +1,175 @@
...
+        if (pipeline.isErrorOccurred) {
+          command.error(
+            buildPipelineErrorMessage(
+              `❌ An error occurred during building process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:39]: An error occurred during the building process

File: cli/src/utils.ts [hunk 24]
```diff
@@ -1,176 +1,175 @@
...
+              pipeline.comment
+            )
+          );
         }
-        await new Promise((resolve) => setTimeout(resolve, 3000));
+        break;
+      case DeployPipelineStatusEnum.IMAGE_BUILDING:
+        CliUx.ux.action.start('◷ Building your squid');
```
Comment (DLangellotti) [inline cli/src/utils.ts:46]: Building your Squid

File: cli/src/utils.ts [hunk 25]
```diff
@@ -1,176 +1,175 @@
...
+        if (pipeline.isErrorOccurred) {
+          command.error(
+            buildPipelineErrorMessage(
+              `❌ An error occurred during building process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:50]: An error occurred during the building process

File: cli/src/utils.ts [hunk 26]
```diff
@@ -1,176 +1,175 @@
...
+              pipeline.comment
+            )
+          );
+        }
+        break;
+      case DeployPipelineStatusEnum.IMAGE_PUSHING:
+        CliUx.ux.action.start('◷ Publishing your squid');
```
Comment (DLangellotti) [inline cli/src/utils.ts:57]: Publishing your Squid

File: cli/src/utils.ts [hunk 27]
```diff
@@ -1,176 +1,175 @@
...
+        if (pipeline.isErrorOccurred) {
+          command.error(
+            buildPipelineErrorMessage(
+              `❌ An error occurred during pushing process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:61]: An error occurred during the pushing process

File: cli/src/utils.ts [hunk 28]
```diff
@@ -1,176 +1,175 @@
...
+            )
+          );
+        }
+        break;
+      case DeployPipelineStatusEnum.DEPLOYING:
+        CliUx.ux.action.start('◷ Almost ready');
+        if (pipeline.isErrorOccurred) {
+          command.error(
+            buildPipelineErrorMessage(
+              `❌ An error occurred during deploying process`,
```
Comment (DLangellotti) [inline cli/src/utils.ts:72]: An error occurred during deployment

File: cli/src/utils.ts [hunk 29]
```diff
@@ -1,176 +1,175 @@
...
+              pipeline.comment
+            )
+          );
+        }
+        break;
+      case DeployPipelineStatusEnum.OK:
+        command.log(
+          `◷ Your squid almost ready and will be accessible on ${deploymentUrl}`
```
Comment (DLangellotti) [inline cli/src/utils.ts:80]: Your Squid is almost ready and will be accessible at

File: cli/src/utils.ts [hunk 30]
```diff
@@ -1,176 +1,175 @@
...
-            values: remotes.map((remote) => remote.name),
-        }).catch(() => {
-            command.error('Canceled', { code: '1' });
-        });
-        remoteUrl = remotes.find(
-            (remote) => remote.name === selected.value
-        ) as RemoteWithRefs;
-    }
-    await git.listRemote([remoteUrl.name]).catch(() => {
-        command.error(`Remote url with name ${remoteUrl.name} not exists`, {
```
Comment (DLangellotti) [inline cli/src/utils.ts:117]: Remote URL with name... does not exist

File: cli/src/utils.ts [hunk 31]
```diff
@@ -1,176 +1,175 @@
...
-        `${remoteUrl.name}/${branch}`,
-    ] as LogOptions<DefaultLogFields>);
-    if (
-        !localCommit.latest ||
+    remoteUrl = remotes.find(
+      (remote) => remote.name === selected.value
+    ) as RemoteWithRefs;
+  }
+  await git.listRemote([remoteUrl.name]).catch(() => {
+    command.error(`Remote url with name ${remoteUrl.name} not exists`, {
```
Comment (DLangellotti) [inline cli/src/utils.ts:116]: Remote URL with name ... does not exist

File: cli/src/utils.ts [hunk 32]
```diff
@@ -1,176 +1,175 @@
...
+  if (status.files && status.files.length) {
+    command.error(`There are unstaged or uncommitted changes`);
+  }
+  await git.fetch();
+  const remoteBranchRefs = await git.listRemote([
+    `${remoteUrl.name}`,
+    `${branch}`,
+  ]);
+  if (remoteBranchRefs === '') {
+    command.error(`Remote branch "${remoteUrl.name}/${branch}" not exists`);
```
Comment (DLangellotti) [inline cli/src/utils.ts:131]: Remote branch ... does nto exist

**Comment by eldargab** [comment]
Please, remove eslint.

1. It's not a good practice to mix such massive formatting changes with regular patches.

2. Suggested code style rules contradict the entire `squid` codebase. Let's discuss it first.

**Pull request closed without merging**